src/main/java/net/snowflake/client/core/SFResultSetMetaData.java

```diff
@@ -403,6 +415,14 @@ public int getPrecision(int column) {
     }
   }
 
+  public int getDimension(int column) {
+    if (dimensions != null && dimensions.size() >= column) {
```

Comment:
If you check size, maybe `> 0` is also worth having?

Reply:
added check